enable.ps1

```diff
@@ -148,151 +69,379 @@ function Resolve-ZenyaErrorMessage {
+    
+                    $updatedAccount = Invoke-RestMethod @splatWebRequest -Verbose:$false
+                    # Set the correct account reference
+                    $outputContext.AccountReference = [PSCustomObject]@{
+                        Id       = $updatedAccount.id
+                        Username = $updatedAccount.userName
+                    }
+    
+                    $outputContext.AuditLogs.Add([PSCustomObject]@{
+                            # Action  = "" # Optional
+                            Message = "Successfully updated account [$($currentAccount.Username)]. AccountReference: $($outputContext.AccountReference | ConvertTo-Json -Depth 10). Old values: $($changedPropertiesObject.oldValues | ConvertTo-Json -Depth 10). New values: $($changedPropertiesObject.newValues | ConvertTo-Json -Depth 10)"
+                            IsError = $false
+                        })
+                }
+                else {
+                    Write-Warning "DryRun: Would update account [$($currentAccount.Username)]. AccountReference: $($actionContext.References.Account | ConvertTo-Json -Depth 10). Old values: $($changedPropertiesObject.oldValues | ConvertTo-Json -Depth 10). New values: $($changedPropertiesObject.newValues | ConvertTo-Json -Depth 10)"
+                }
+                break
+            }
+            catch {
+                $ex = $PSItem
+                $errorMessage = Get-ErrorMessage -ErrorObject $ex
+    
+                Write-Verbose "Error at Line [$($ex.InvocationInfo.ScriptLineNumber)]: $($ex.InvocationInfo.Line). Error: $($errorMessage.VerboseErrorMessage)"
+                Write-Verbose "URI: $($splatWebRequest.Uri)"
+                Write-Verbose "Body: $($body)"
+    
+                $outputContext.AuditLogs.Add([PSCustomObject]@{
+                        # Action  = "" # Optional
+                        Message = "Error updating account [$($currentAccount.Username)]. AccountReference: $($actionContext.References.Account | ConvertTo-Json -Depth 10). Error Message: $($errorMessage.AuditErrorMessage)"
+                        IsError = $true
+                    })
+    
+                # Throw terminal error
+                throw
+            }
+        }
+        'NoChanges' {
+            if (-not($dryRun -eq $true)) {
+                Write-Verbose "Skipped updating account [$($account.Username)]. Reason: No changes"
+
+                # Set the correct account reference
+                $outputContext.AccountReference = [PSCustomObject]@{
+                    Id       = $currentAccount.id
+                    Username = $currentAccount.userName
+                }
+            }
+            else {
+                Write-Warning "DryRun: Would skip updating account [$($account.Username)]. Reason: No changes"
+            }
+            break
+        }
     }
-} 
-Write-Output $result | ConvertTo-Json -Depth 10
\ No newline at end of file
+}
+catch {
+    $ex = $PSItem
```

Comment:
do we log this in the auditlogs?

Reply:
No, because it is already logged as auditlog if the action fails.
This catch is purely for dryrun.